retry

```diff
@@ -37,14 +37,27 @@ done
 retries="${retries:-3}"
 sleep="${sleep:-3}"
 
-ret=0
-until "$@"; do
+run () {
+  # prepare command
+  cmd=""
+  for param in "${@}"; do cmd="$cmd '$param'"; done
```

Comment:
I think this is asking for trouble:
```
./retry -r 2 -s 1 -- echo "'" 
```
Not sure why you would turn all args into a string and then execute that instead of just executing the array like it was done before?